framework/assets/yii.validation.js

```diff
@@ -13,7 +13,7 @@
 yii.validation = (function ($) {
     var pub = {
         isEmpty: function (value) {
-            return value === null || value === undefined || value == [] || value === '';
+            return value === null || value === undefined || ($.isArray(value) && value.length === 0) || value === '';
```

Comment:
Fixed bug. 

In JavaScript array can not be determined just by using this kind of check: `value == []`, `[] == []` is always `false` for example. Since the code is jQuery dependent it's OK to use `$.isArray` (and by the way it's used above in the same file). Another advantage of using `$.isArray` over [Array.isArray()](https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Global_Objects/Array/isArray) is that 1st version branch has fallback for older browsers, so in case user want that he need just override jquery version in `yii\web\JQueryAsset`.